### PR TITLE
cli: addition of reindex command

### PR DESCRIPTION
**Pull request by jirikuncar**

### CODE REVIEW

File: invenio_indexer/cli.py [hunk 1]
```diff
@@ -28,27 +28,52 @@
...
+
+
+@index.command()
+@click.option('--yes-i-know', is_flag=True, callback=abort_if_false,
+              expose_value=False,
+              prompt='Do you really want to reindex all records?')
+@with_appcontext
+def reindex():
+    """Reindex all records."""
+    click.secho('Sending records to indexing queue ...', fg='yellow')
```
Comment (lnielsen) [inline invenio_indexer/cli.py]: What about using ``fg=green`` as the task is doing what it's supposed to do.

File: invenio_indexer/cli.py [hunk 2]
```diff
@@ -28,27 +28,52 @@
...
+        click.secho('Indexing records...', fg='green')
         RecordIndexer().process_bulk_queue()
+
+
+@index.command()
+@click.option('--yes-i-know', is_flag=True, callback=abort_if_false,
+              expose_value=False,
+              prompt='Do you really want to reindex all records?')
+@with_appcontext
+def reindex():
```
Comment (lnielsen) [inline invenio_indexer/cli.py]: Not sure if we should add this in the help text, or as a ``click.secho``, but perhaps something along the lines of:

``INFO: Deleted records are not removed from the index.``

File: invenio_indexer/cli.py [hunk 3]
```diff
@@ -28,27 +28,52 @@
...
+    """Reindex all records."""
+    click.secho('Sending records to indexing queue ...', fg='yellow')
+
+    def records():
+        """Record iterator."""
+        for record in RecordMetadata.query.values(RecordMetadata.id):
+            yield record[0]
+
+    RecordIndexer().bulk_index(records())
+    click.secho('Execute "run" command to process the queue!', fg='red')
```
Comment (lnielsen) [inline invenio_indexer/cli.py]: Same here, perhaps either no color or use of yellow/green. For me red means that something produced an error, which is not the case.

**Comment by lnielsen** [comment]
LGTM , besides the cosmetics.

**Comment by jirikuncar** [comment]
@lnielsen I have amended the commit.

File: invenio_indexer/cli.py [hunk 4]
```diff
@@ -28,27 +28,55 @@
...
+    NOTE: Deleted records are not removed from the index.
+    """
+    click.secho('Sending records to indexing queue ...', fg='green')
+
+    def records():
+        """Record iterator."""
+        for record in RecordMetadata.query.values(RecordMetadata.id):
+            yield record[0]
+
+    RecordIndexer().bulk_index(records())
```
Comment (hachreak) [inline invenio_indexer/cli.py]: You can replace lines from 76 to 81 with:
```python
RecordIndexer().bulk_index([r[0] for r in RecordMetadata.query.values(RecordMetadata.id)])
```

Comment (jirikuncar) [inline invenio_indexer/cli.py]: I prefer to keep explicit iterator over building full list of UUIDs. I could replace it with:

```python
RecordIndexer().bulk_index(r[0] for r in RecordMetadata.query.values(RecordMetadata.id))
```

Then I should break long line hence:

```python
RecordIndexer().bulk_index(
    r[0] for r in RecordMetadata.query.values(RecordMetadata.id)
)
```

The truth is I will save some lines, but later we want to introduce some filtering, hence the `RecordMetadata.query` part will have to be extended and reading from arguments or `stdin` will need to be added.

At the end, I can rewrite it to

```python
records  = (r[0] for r in RecordMetadata.query.values(RecordMetadata.id))
RecordIndexer().bulk_index(records)
```

Does it make it more readable? Maybe yes. I should definitely add a comment so nobody refactors the iterator to a list comprehension.

```python
# NOTE records should be an iterator to save memory.
records  = (r[0] for r in RecordMetadata.query.values(RecordMetadata.id))
RecordIndexer().bulk_index(records)
```

**Pull request merged**